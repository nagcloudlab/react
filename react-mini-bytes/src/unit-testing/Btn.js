import React, { useState } from "react";

function Btn({value=0}) {
  const [count, setCount] = useState(value);
  return (
    <div>
      <button title="counter" onClick={(e) => setCount((prev) => prev + 1)} className="btn btn">
        {count}
      </button>
    </div>
  );
}

export default Btn;
