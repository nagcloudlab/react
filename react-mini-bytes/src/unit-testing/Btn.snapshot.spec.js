
import renderer from 'react-test-renderer';
import Btn from './Btn'

it('renders correctly', () => {
    const tree = renderer.create(<Btn />).toJSON();
    expect(tree).toMatchSnapshot();
});