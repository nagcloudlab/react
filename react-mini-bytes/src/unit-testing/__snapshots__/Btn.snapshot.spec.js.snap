// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <button
    className="btn btn"
    onClick={[Function]}
    title="counter"
  >
    0
  </button>
</div>
`;
